### DISCUSSION TITLE
cx_Freeze completes without errors but empty build directory

### DISCUSSION THREAD

**Discussion by gcailly** [marcelotduarte/cx_Freeze#2087] [Q&A]
Hello !
I have a problem :

I'm attempting to freeze my Python application using cx_Freeze. The process seems to complete without any errors, but the resulting build directory is empty.

Project Structure:
.
```
├── data
│   └── ... (various assets)
├── docs
│   └── ... (documentation files)
├── md_editor
│   ├── bold.png
│   ├── ...
│   ├── md_editor.py
│   └── __init__.py
├── translations
│   └── ... (translation files)
├── utils
│   ├── custom_msg.py
│   ├── widgets.py
│   └── __init__.py
├── main.py
└── setup.py
```

In main.py, I import the md_editor module as:

`from md_editor.md_editor import MdEditorWidget`

Inside md_editor.py, I import the utils module as:

`from utils.custom_msg import CustomMsgBox`

Issue:

When I run the cx_Freeze command using my setup.py, I only receive the following output:

```
INFO:root:running build
INFO:root:running build_exe
```

There are no error messages, but when I check the build directory, it's empty.

Things I've Tried:

- Ensured all __init__.py files are present in my custom module directories.
- Verified the application runs correctly without freezing.
- Checked the cx_Freeze documentation and FAQ but couldn't find a solution to my specific problem.

I'm at a loss as to why the build directory remains empty without any error messages. Any insights or suggestions would be greatly appreciated!

**Comment by marcelotduarte** [comment]
Assuming your app runs with `python main.py`
Can you test some samples (advanced, asmodule, relimport and some other) at https://github.com/marcelotduarte/cx_Freeze/tree/main/samples to check your installation.
Alias, what is your SO, python version (python -VV), cx_Freeze version?

**Reply by gcailly** [reply]
The app runs with python in a virtual environment.
I'm sorry, how can I test the samples ? With a setup.py trying to freeze the .py files in the samples directory you gave ?
SO : Windows 11 22H2
Python 3.11.5
cx_Freeze 6.15.9

**Comment by marcelotduarte** [comment]
> The app runs with python in a virtual environment.

But the command to run is python main.py?

Are you using setup? Can you put the code of setup.py here?

> I'm sorry, how can I test the samples ? With a setup.py trying to freeze the .py files in the samples directory you gave ?

Download the directory and run the sample with `python setup.py build` in that directory and then check if it runs at build/exe....../test_.....
The .... should be replaced accordling.
